### PR TITLE
Split libolaserver into two parts to stop dpkg-shlibdeps warnings

### PR DESCRIPTION
**Pull request by peternewman**
If you're happy with this, I'll revert the revert of plugin commits so it's actually makes a difference to the build and stops the errors.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -132,6 +132,7 @@ include include/Makefile.mk
 include javascript/Makefile.mk
 include man/Makefile.mk
 include ola/Makefile.mk
+include olad/plugin_api/Makefile.mk
```
Comment (nomis52) [inline Makefile.am]: Add a comment here as to why this is included at the top level. Normally we just include the Makefiles in the first level of directories.

File: olad/Makefile.mk [hunk 2]
```diff
@@ -63,14 +43,7 @@ ola_server_additional_libs += common/http/libolahttp.la
...
 # lib olaserver
-lib_LTLIBRARIES += olad/libolaserverplugininterface.la olad/libolaserver.la
-
-olad_libolaserverplugininterface_la_SOURCES = \
-    $(ola_server_plugin_interface_sources)
-olad_libolaserverplugininterface_la_CXXFLAGS = $(COMMON_CXXFLAGS)
-olad_libolaserverplugininterface_la_LIBADD = common/libolacommon.la \
-                                             common/web/libolaweb.la \
-                                             ola/libola.la
+lib_LTLIBRARIES += olad/plugin_api/libolaserverplugininterface.la olad/libolaserver.la
```
Comment (nomis52) [inline olad/Makefile.mk]: olad/plugin_api/libolaserverplugininterface.la should be done in olad/plugin_api/Makefile.am

File: olad/plugin_api/Makefile.mk [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    olad/plugin_api/PortBroker.cpp \
+    olad/plugin_api/PortManager.cpp \
+    olad/plugin_api/PortManager.h \
+    olad/plugin_api/Preferences.cpp \
+    olad/plugin_api/Universe.cpp \
+    olad/plugin_api/UniverseStore.cpp \
+    olad/plugin_api/UniverseStore.h
+
+olad_plugin_api_libolaserverplugininterface_la_SOURCES = \
+    $(ola_server_plugin_interface_sources)
```
Comment (nomis52) [inline olad/plugin_api/Makefile.mk]: Just add the list of sources here, we don't need the intermediate variable now

File: olad/plugin_api/Makefile.mk [hunk 4]
```diff
@@ -0,0 +1,77 @@
+# This file is included directly in Makefile.am rather than via
+# olad/Makefile.mk due to the order of dependencies between them; we need to
+# build olad/plugin_api, then the plugins, then olad
+
+# HEADERS
+##################################################
+noinst_HEADERS += olad/plugin_api/TestCommon.h
+
+# LIBRARIES
```
Comment (nomis52) [inline olad/plugin_api/Makefile.mk]: Add 

lib_LTLIBRARIES += olad/plugin_api/libolaserverplugininterface.la

File: olad/Makefile.mk (hunk 2 above)
Comment (peternewman) [inline olad/Makefile.mk]: Done.

File: olad/plugin_api/Makefile.mk (hunk 3 above)
Comment (peternewman) [inline olad/plugin_api/Makefile.mk]: Done.

File: olad/plugin_api/Makefile.mk (hunk 4 above)
Comment (peternewman) [inline olad/plugin_api/Makefile.mk]: Done.

File: Makefile.am (hunk 1 above)
Comment (peternewman) [inline Makefile.am]: Done.

**Comment by nomis52** [comment]
This looks fine. Do you want to add the pluginapi lib to all the plugins now?

**Comment by peternewman** [comment]
Done.

**Comment by nomis52** [comment]
LGTM

**Pull request merged**